src/command.ts

```diff
@@ -308,3 +310,22 @@ export class Command {
     };
   }
 }
+
+// Project IDs must follow a certain format, as documented at:
+// https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects#resource:-project
+// However, the regex below, matching internal ones, is more permissive so that
+// some legacy projects with irregular project IDs still works.
+const PROJECT_ID_REGEX = /^(?:[^:]+:)?[a-z0-9\-]+$/;
+
+export function validateProjectId(project: string): void {
```

Comment:
Please add a jsdoc block and include that it throws if it is invalid

```typescript
/**
 * Description...
 * @param project the project ID to check.
 */
```

Reply:
Done.